bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -216,6 +248,26 @@ final class BagsManager(executorService: ExecutorService) {
         case _                             => // Ignore
       }
   }
+
+  /** BAGS status changes. **/
+  def bagsStatus(key: SPNodeKey): Option[BagsStatus] =
+    synchronized(state.statuses.get(key))
+
+  /** Listeners for BAGS status changes. **/
+  private var listeners: List[BagsStatusListener] = Nil
```

Comment:
Since the `List` itself is immutable if we can get a reference to it we don't have to worry about it changing out from under us at least.  My concern was more with holding a lock while calling listeners.

It does seem like the add/remove/clear methods below should be synchronized though.  Do we need to support arbitrary lists of listeners? If there is only one interested party perhaps the `BagsManager` could just be created with a callback function to receive status updates.